### PR TITLE
ref(alerts): Remove crash rate v1 code

### PR DESCRIPTION
**Pull request by ceorourke**
Remove the crash rate v1 code that was replaced by the v2 code [2 years ago](https://github.com/getsentry/sentry/pull/34957). I confirmed with the ingest team that they're not using this anymore and it's safe to remove. 

Closes https://getsentry.atlassian.net/browse/ALRT-308

### CODE REVIEW

File: src/sentry/incidents/subscription_processor.py
Comment (saponifi3d) [inline src/sentry/incidents/subscription_processor.py:361]: nit: kinda like how the variables were defined at the top where possible :)

**Review by saponifi3d — APPROVED** [review]

File: src/sentry/incidents/subscription_processor.py
Comment (saponifi3d) [inline src/sentry/incidents/subscription_processor.py:360]: Looks like there's a test case from somewhere not being met: https://github.com/getsentry/sentry/actions/runs/11075656230/job/30777097641?pr=78275

Comment (ceorourke) [reply src/sentry/incidents/subscription_processor.py:360]: I don't have a way to store crash rate data locally, but everything I removed was explicitly labeled as such (including tests). I'll let the ingest team approve before I merge though since they have the most context.

File: tests/sentry/incidents/test_subscription_processor.py
Comment (jjbayer) [inline tests/sentry/incidents/test_subscription_processor.py:3541]: Shouldn't this still happen on V2?

**Review by jjbayer — APPROVED** [review]

File: tests/sentry/incidents/test_subscription_processor.py
Comment (ceorourke) [reply tests/sentry/incidents/test_subscription_processor.py:3541]: V2 doesn't go down the same code path - V1 called [this function](https://github.com/getsentry/sentry/blob/master/src/sentry/snuba/entity_subscription.py#L378) and the `MetricIndexNotFound` error was raised [here](https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_metrics/utils.py#L88-L96). Is that expected?

Comment (ceorourke) [reply tests/sentry/incidents/test_subscription_processor.py:3541]: This made me realize nothing calls `translate_sessions_tag_keys_and_values` anymore so I can remove that too.

Comment (jjbayer) [reply tests/sentry/incidents/test_subscription_processor.py:3541]: Yes, makes sense. Thanks for cleaning this up!

**Pull request merged**
